pgjdbc/src/main/java/org/postgresql/jdbc/PgArray.java

```diff
@@ -310,14 +316,40 @@ private PgArrayList buildArrayList(String fieldString) throws SQLException {
     }
   }
 
+  private TimestampUtils getTimestampUtils() {
+    if (timestampUtils == null) {
```

Comment:
can you use the resource lock here. see https://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html

Reply:
Done.